### ISSUE TITLE
typedef with default param

### ISSUE THREAD

**Issue by testerez** [palantir/tslint#2194]
### Bug Report

- __TSLint version__: `4.3.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: CLI  / Visual Studio

#### TypeScript code being linted
```ts
function foo(value = true) {
  // ...
}
```

#### Actual behavior

> [tslint] expected parameter: 'value' to have a typedef (typedef)

#### Expected behavior

As the type of `value` is infered from the default param, I'd like tslint to consider it valid.

**Comment by r-ohan** [comment]
Having the same issue.

**Comment by nchen63** [comment]
This issue isn't just with params. I see this as being a new option `allow-assignment`

**Comment by adidahiya** [comment]
this is basically a duplicate of #669.
also related: #813.
